### PR TITLE
VDA-ERC20 : Update for transferring ownership

### LINKED ISSUE #89: [VDA-ERC-20] Update transferring ownership feature
- Need to give `mint` role to the new owner

### PR DESCRIPTION
**Pull request by ITStar10**
Updated for transferOwnership
	- Mint & Adding minter role to the new owner
	- Revoke the Mint & Adding minter role from the original owner

Updated for renounceOwnership
	- Revoke the Mint & Adding minter role from the original owner

### CODE REVIEW

File: VDA-ERC20/contracts/VDA-V1.sol [hunk 1]
```diff
@@ -99,9 +99,8 @@ contract VeridaToken is ERC20PausableUpgradeable, OwnableUpgradeable,
     /**
      * @dev Burn `amount` tokens from `to`.
      */
-    function burn(address from, uint256 amount) external override {
```
Comment (tahpot) [inline VDA-ERC20/contracts/VDA-V1.sol:102]: @ITStar10 Why was the ability to burn tokens for a specific address removed?

File: VDA-ERC20/contracts/VDA-V1.sol [hunk 2]
```diff
@@ -280,4 +279,23 @@ contract VeridaToken is ERC20PausableUpgradeable, OwnableUpgradeable,
 
         isTransferEnabled = true;
     }
+
+    /**
+     * See {OwnableUpgradeable.sol}
+     * @dev give `mint` role to the new owner
+     */
+    function _transferOwnership(address newOwner) internal virtual override {
```
Comment (tahpot) [inline VDA-ERC20/contracts/VDA-V1.sol:287]: Where is this function called?

File: VDA-ERC20/contracts/VDA-V1.sol (hunk 1 above)
Comment (ITStar10) [reply VDA-ERC20/contracts/VDA-V1.sol:102]: Just `burn` function was created by copy & paste the `mint(...)` function.
It's now updated to as following:
```
function burn(uint256 amount) external;
```
It's burning the token of function caller now.

File: VDA-ERC20/contracts/VDA-V1.sol (hunk 2 above)
Comment (ITStar10) [reply VDA-ERC20/contracts/VDA-V1.sol:287]: This function is defined in the `openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol`. And it's called when the contract owner transfer or renounce the ownership.
In this function, we're revoking the `mint` & `Add/Revoke minters` permission from old owner and adding these features to the new owner.

File: VDA-ERC20/contracts/VDA-V1.sol (hunk 1 above)
Comment (tahpot) [reply VDA-ERC20/contracts/VDA-V1.sol:102]: How would this be used?

File: VDA-ERC20/contracts/VDA-V1.sol (hunk 2 above)
Comment (tahpot) [reply VDA-ERC20/contracts/VDA-V1.sol:287]: Got it, thanks

File: VDA-ERC20/contracts/VDA-V1.sol (hunk 1 above)
Comment (ITStar10) [reply VDA-ERC20/contracts/VDA-V1.sol:102]: Any token holders can burn their token by this function.
Of course, normal users don't feel to burn their tokens.
By the way, token  manager - Verida need to burn tokens for several reasons.
If you think this is not necessary, we can remove this function.

Comment (ITStar10) [reply VDA-ERC20/contracts/VDA-V1.sol:102]: Once any token holders burn their tokens, the total supply of the Verida Token will be decreased.

**Pull request merged**